### DISCUSSION TITLE
Support for Material-UI V5

### DISCUSSION THREAD

**Discussion by MDrooker** [vazco/uniforms#1011] [Q&A] [answered]
Material-UI V5 is out.
npm@latest points to v5.0.0-rc.0

The namespace has change to '@mui/material'.
So the uniforms-material doesnt find it.

Any thoughts around supporting V5?

**Comment by wadamek65** [comment, marked as answer]
Hey @MDrooker ,

It seems like v5 is still a release candidate so it's not treated as stable so we'll not support it just yet. If you'd like to use it in your project already, you can always create your own custom theme instead of using `uniforms-material`. You can find the documentation for it [here](https://uniforms.tools/docs/tutorials-creating-custom-theme) except in this case you will most likely want to copy the `uniforms-material` package instead of `uniforms-unstyled`.

Once stable v5 is released we will have to investigate what exactly the update will entail and will discuss further details about supporting it. We will have to see whether we will want to update the current theme or create a new one like with `bootstrap` so until then I don't think we can give any more concrete information. You are always welcome to create an issue/PR for this yourself once it is stable.

**Reply by MDrooker** [reply]
Sounds good.  Thanks for the look.  This package really makes a ton of peoples lives easier.  So thank you.

**Reply by radekmie** [reply]
@wadamek65, @MDrooker: are you up-to-date with the breaking changes that Material-UI v5 brings in? I've skimmed through the [migration guide](https://next.material-ui.com/guides/migration-v4/) and it doesn't seem _that much_ (from our perspective), except for the package name change, that'd force us to do a breaking change.

**Reply by wadamek65** [reply]
@radekmie I haven't researched the changes more in-depth yet so can't say more than you know already.

**Reply by MDrooker** [reply]
Sorry for the delay-
Other than the name space change, there were some changes with using the emotion engine for styling.
The reset was seamless.

They also released 5.0 today-
https://github.com/mui-org/material-ui/releases/tag/v5.0.0

**Reply by MDrooker** [reply]
Following back on this..
Any thoughts on a new V5 package?  Or just upgrading the current material-ui one?

**Reply by radekmie** [reply]
@wadamek65, @MDrooker I've looked into it a little bit more. Because they moved to a different package name, we're not able to support both. Because of that, we'll have to create a new theme. There's no ETA for that, though.

**Reply by MDrooker** [reply]
Not to be that guy....but what are the thoughts for moving this forward?

**Reply by wadamek65** [reply]
@MDrooker As it isn't one of our highest priorities right now, there is still no ETA. We're open to any contributions though and will help with the PR as much as we can.
